### ISSUE TITLE
bug or docs: useHeadSafe and useServerHeadSafe strips href on link

### ISSUE THREAD

**Issue by schalkneethling** [unjs/unhead#162]
### Environment

Using the bundled version that comes with Nuxt 3.5.1

### Reproduction

Add the following to `app.vue` inside the script setup (`<script lang="ts" setup>`):

```js
useServerHeadSafe({
  link: [
    {
      href: "/favicon.svg", // this file exists inside the `/public` directory
      rel: "icon",
      type: "image/svg+xml",
    },
  ],
});
```

When you open the page in a browser, the `rel` and `type` attributes are present, but the `href` attribute is dropped.

> **Note:** The same happens when using `useHeadSafe`

### Describe the bug

According to the documentation, the `href` property on `link` is in the allowed list.
https://unhead.harlanzw.com/guide/composables/use-head-safe#logic

As mentioned above, when loaded in the browser, the `href` attribute is not present on the `link` element.

**Comment by harlan-zw** [comment]
This is a bug, it should be implemented to allow href on a subset of rels.

Will try get to it when I have a chance, PRs welcome 🙂

**Comment by harlan-zw** [comment]
This should be fixed in v1.1.28, let me know if you have any issues :)
